### ISSUE TITLE
Feature: Support Tools for Huggingface models

### ISSUE THREAD

**Issue by andreas-eberle** [quarkiverse/quarkus-langchain4j#102]
I just tried to use Tools with Huggingface, but it seems that is not yet supported. I got this error:

```
 java.lang.IllegalArgumentException: Tools are currently not supported for HuggingFace models
	at io.quarkiverse.langchain4j.huggingface.QuarkusHuggingFaceChatModel.generate(QuarkusHuggingFaceChatModel.java:108)
	at dev.langchain4j.model.chat.ChatLanguageModel_XNMsOaekknG7BdNZ5YSUkjh1SqE_Synthetic_ClientProxy.generate(Unknown Source)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupport.implement(MethodImplementationSupport.java:104)
	at com.arconsis.youtube.quarkus.langchain.services.ai.MyAiService$$QuarkusImpl.findUserLocations(Unknown Source)
	at com.arconsis.youtube.quarkus.langchain.services.ai.MyAiService_y7si7UWyP2GRrk7fUaA8LGsjFd0_Synthetic_ClientProxy.findUserLocations(Unknown Source)
	at com.arconsis.youtube.quarkus.langchain.rest.EmployeesResource.writePoem(EmployeesResource.java:19)
	at com.arconsis.youtube.quarkus.langchain.rest.EmployeesResource$quarkusrestinvoker$writePoem_71e6bb6015aacbbc37e37d75924cc107b689db1a.invoke(Unknown Source)
	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

Would be great to also have support for tools with Hunggingface.

**Comment by geoand** [comment]
This is not supported in upstream Langchain4j either.

The question is whether the HuggingFace REST API has support for this. If it doesn't, there is nothing we can do

**Comment by andreas-eberle** [comment]
Sure. I didn't kow langchain does not support this and didn't check. I found this section on the huggingface documentation which states the aim to support langchain tools in the future but it's not there yet. 

https://huggingface.co/docs/transformers/custom_tools#future-compatibility-with-langchain

I also now found the [hint in your docs](https://docs.quarkiverse.io/quarkus-langchain4j/dev/huggingface.html#_tools) => closing the issue as it is not a quarkus langchain issue.
